### PR TITLE
Add elixir-credo checker

### LINKED ISSUE #1062: Add checker for elixir credo
This is a reminder issue for integrating https://github.com/aaronjensen/flycheck-credo once #836 is solved.

### PR DESCRIPTION
**Pull request by fapdash**
Transfers https://github.com/aaronjensen/flycheck-credo to flycheck core.
Originally by Aaron Jensen (https://github.com/aaronjensen/)

Closes #1062

### CODE REVIEW

**Comment by fapdash** [comment]
Can't really tell why https://travis-ci.org/flycheck/flycheck/jobs/373665776 and https://travis-ci.org/flycheck/flycheck/jobs/373665777 failed, the tests seem to have all passed?
https://travis-ci.org/flycheck/flycheck/jobs/373665779 failed because travis couldn't reach melpa, should work on a second try.

File: doc/languages.rst [hunk 1]
```diff
@@ -297,6 +297,16 @@ to view the docstring of the syntax checker.  Likewise, you may use
...
 
+.. supported-language:: Elixir
+
+   .. syntax-checker:: elixir-credo
+
+      Check code style with `credo <https://github.com/rrrene/credo>`_
+
+      .. defcustom:: flycheck-elixir-credo-strict
+
+      When non-nil, run credo in strict mode, via ``--strict``.
```
Comment (cpitclaudel) [inline doc/languages.rst]: I think this bit should be indented

File: flycheck.el [hunk 2]
```diff
@@ -7016,6 +7017,39 @@ See URL `http://hadolint.lukasmartinelli.ch/'."
...
+  "Find directory with from which we can run credo."
+  (and buffer-file-name
+       (locate-dominating-file buffer-file-name "deps/credo")))
+
+(flycheck-def-option-var flycheck-elixir-credo-strict nil elixir-credo
+  "Enable strict mode in credo.
+When non-nil, pass the `--strict' flag to credo."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.32"))
```
Comment (cpitclaudel) [inline flycheck.el]: 32, not 0.32

File: flycheck.el [hunk 3]
```diff
@@ -7016,6 +7017,39 @@ See URL `http://hadolint.lukasmartinelli.ch/'."
...
+            "credo"
+            (option-flag "--strict" flycheck-elixir-credo-strict)
+            "--format"
+            "flycheck"
+            "--read-from-stdin"
+            source-original)
+  :standard-input t
+  :predicate
+  (lambda () (and buffer-file-name
+                  (locate-dominating-file buffer-file-name "deps/credo")))
```
Comment (cpitclaudel) [inline flycheck.el]: Isn't that the same as `flycheck-credo--working-directory`?
It should likely be an `enabled` property, too

File: flycheck.el [hunk 4]
```diff
@@ -7016,6 +7017,39 @@ See URL `http://hadolint.lukasmartinelli.ch/'."
...
+  :working-directory flycheck-credo--working-directory
+  :error-patterns
+  ((info line-start (file-name) ":" line ":" column ": " (or "F" "R" "C")  ": " (message) line-end)
+   (info line-start (file-name) ":" line ": " (or "F" "R" "C")  ": " (message) line-end)
```
Comment (cpitclaudel) [inline flycheck.el]: Rather than two patterns, could you wrap the `":" column` part in an optional group?

File: flycheck.el (hunk 3 above)
Comment (fapdash) [reply flycheck.el]: When do you prefer `:enabled` over `:predicate`?
Probably has something to do with this:?
> This property behaves as `:predicate', except that it's only
     called the first time a syntax checker is to be used in a buffer.

Comment (cpitclaudel) [reply flycheck.el]: Correct — there's no need to check for that file on each run.

Comment (fapdash) [reply flycheck.el]: Okay, I've changed it. This now means that you'd have to reopen a buffer after installing credo into a project to active the checker, right?

**Comment by fapdash** [comment]
@cpitclaudel Any idea why he fails to install `elixir-mode` from melpa with emacs 26.1?
https://travis-ci.org/flycheck/flycheck/jobs/374824767

**Comment by cpitclaudel** [comment]
I'm not entirely sure (sorry for the late response) why it's not working.  I *guess* it's an OpenSSL version issue, but I haven't followed the latest developments. @fmdkdd, any idea?

**Comment by fmdkdd** [comment]
Probably #1462 will help.  I have no trouble running `make init` using the Docker images with `elixir-mode` added.

**Comment by cpitclaudel** [comment]
OK, rebased and ready to go.  @fapdash sorry for the long long delay.  @aaronjensen any comments before I merge?

**Comment by aaronjensen** [comment]
Nope

**Comment by cpitclaudel** [comment]
Thanks!

**Pull request merged**